stan/math/rev/fun/singular_values.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef STAN_MATH_REV_FUN_SINGULAR_VALUES_HPP
+#define STAN_MATH_REV_FUN_SINGULAR_VALUES_HPP
+
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/err/check_nonzero_size.hpp>
+#include <Eigen/SVD>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return the singular values of the specified matrix.
+ *
+ * Adjoint update equation comes from Equation (4) in Differentiable Programming
+ * Tensor Networks(H. Liao, J. Liu, et al., arXiv:1903.09650).
+ *
+ * @tparam EigMat type of input matrix
+ * @param m MxN input matrix
+ * @return Singular values of matrix
+ */
+template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr,
```

Comment:
Change: `require_eigen_matrix_dynamic_t<EigMat>` to `require_rev_matrix_t<EigMat>`

That means accept `Eigen::Matrix<var, R, C>` and `var_value<Eigen::Matrix<double, R, C>>` types, not just the first.

Delete `require_vt_var<EigMat>`. Not necessary.